### ISSUE TITLE
Expected token errors should display all expected tokens, not just the first

### ISSUE THREAD

**Issue by sandersn** [microsoft/TypeScript#43467]
### 🕗 Version & Regression Information

Pretty sure this has been the behaviour since the rewrite.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ssl=2&ssc=25&pln=1&pc=1#code/AQMwrgdgxgLglgewsAhgBgBQG0UBpgBG+WWUAuhQFzBYRgC2BApgE752Ov4DOMLcEAOZYyIsgEpgAb2ABfAFCpMWAIz4ATMSwAiAO4IWAGwAm2iuIDcQA)

### 💻 Code
```ts
function a0([a, b, [[c]]]: [number, number, string[][]]) { }
a0([1, 2, [["world"]]);
```

### 🙁 Actual behavior

On the last `)`, "Expected ','."

### 🙂 Expected behavior

On the last `)`, "Expected ',' or ']'."

Right now the parser only shows the first error that starts at each location. This is an OK approach, especially when parsing has gone badly wrong, but often the parser will expect multiple tokens. The parser should report all the tokens in one error that were previously issued as multiple Expected_0 errors.
